File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSChangeTrackStateRequestExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.sdk.models.trackchangerequest.HMSChangeTrackStateRequest

class HMSChangeTrackStateRequestExtension {

    companion object{
        fun toDictionary(hmsChangeTrackStateRequest: HMSChangeTrackStateRequest?):HashMap<String,Any?>?{
            val hashMap = HashMap<String,Any?>()
            if(hmsChangeTrackStateRequest==null)return null
            hashMap["mute"] = hmsChangeTrackStateRequest.mute
            hashMap["requested_by"] = HMSPeerExtension.toDictionary(hmsChangeTrackStateRequest.requestedBy)
            hashMap["track"] = HMSTrackExtension.toDictionary(hmsChangeTrackStateRequest.track)

            val args=HashMap<String,Any?>()
            args["track_change_request"] = hashMap

            return  args
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/Models/HMSChangeTrackStateRequestExtension.swift
//
//  HMSChangeTrackStateRequestExtension.swift
//  hmssdk_flutter
//
//  Created by Yogesh Singh on 06/01/22.
//

import Foundation
import HMSSDK

class HMSChangeTrackStateRequestExtension {

    static func toDictionary(_ request: HMSChangeTrackStateRequest) -> [String: Any] {

        var dict = [String: Any]()

        dict["mute"] = request.mute

        dict["track"] = HMSTrackExtension.toDictionary(request.track)

        if let peer = request.requestedBy {
            dict["requested_by"] = HMSPeerExtension.toDictionary(peer)
        }

        return dict
    }
}
